File: candid/tests/assets/codegen/basic/recursive.js
export default function({ IDL }) {
  const List = IDL.Record({
    'head': IDL.Nat32,
    'tail': List,
  });

  const DoubleRecursion = IDL.Record({
    'a': DoubleRecursionA,
  });

  const DoubleRecursionA = IDL.Opt(DoubleRecursion);

}

File: rust/candid_parser/tests/assets/codegen/basic/recursive.rs
#[derive(Clone)]
pub struct List {
    pub head: u32,
    pub tail: Box<List>,
}
#[derive(Clone)]
pub struct DoubleRecursion {
    pub a: DoubleRecursionA,
}
pub type DoubleRecursionA = Option<Box<DoubleRecursion>>;
